pimd/pim_cmd_common.c

```diff
@@ -566,6 +566,17 @@ int pim_process_rp_cmd(struct vty *vty, const char *rp_str,
 		return CMD_WARNING_CONFIG_FAILED;
 	}
 
+#if PIM_IPV == 4
+	if ((pim_addr_is_any(rp_addr)) || (pim_addr_is_multicast(rp_addr)))
```

Comment:
I think it's better to define pim_addr_is_multicast api as v4 and v6 specific, then you don't need to have #if PIM_IPV == 4 and else checks and duplicate code here as well as in the api.
That way you can just call like this without the if else checks:
if ((pim_addr_is_any(rp_addr)) || (pim_addr_is_multicast(rp_addr)))

You can rename pim_addr_is_multicast api accordingly.

Reply:
Thanks @mobash-rasool. I have updated the patch